app.py

```diff
@@ -1,45 +1,120 @@
+def remove_duplicate_message(items):
+    """
+    Remove duplicate messages from the list
+    """
+
+    # parse the JSON strings and create a list of dictionaries
+    parsed_items = [json.loads(item) for item in items]
+
+    # Use another list comprehension to create a list of unique dictionaries
+    filtered_items = [
+        dict(i) for i in set(tuple(i.items()) for i in parsed_items)  # noqa: E501
+    ]
+
+    return filtered_items
+
+
+def archive_files(bucket, all_keys, dest="archive"):
     """
     Archive the processed file to avoid possible scenarios of race conditions.
     We currently use meta.client.copy instead of client.copy b/c it can copy
     multiple files via multiple threads, since we have batching in view.
 
     Args:
         bucket (str): The s3 bucket name
-        key (str): The key of the file to be archived
+        all_keys (list): The key of the file to be archived
+        dest (str): The destination folder to move the file to
     Returns:
         None
 
     Example:
         >>> archive_processed_file("my-bucket", "my-file.json")
     """
 
-    global s3_resource
+    for key in all_keys:
+        copy_source = {"Bucket": bucket, "Key": key}
+
+        logger.info(f"Archiving file {key} to {dest}")
+        logger.info(f"copy_source: {copy_source}")
+
+        try:
+            s3_resource.meta.client.copy(copy_source, bucket, f"{dest}/{key}")
+        except Exception as e:
+            logger.error(f"Error archiving file {key}: {e}")
+            continue
+        logger.info(f"Archived file :: {key}")
+
+        # delete object from original bucket
+        s3_client.delete_object(Bucket=bucket, Key=key)
+        logger.info(f"Deleted file :: {key}")
+
+
+def retrieve_messages_from_queue():
+    """
+    Get the file paths from the SQS.
+    """
+
+    # Create SQS client
+    sqs = boto3.client("sqs")
+
+    sqs_url = os.environ.get("SQS_QUEUE_URL")
+
+    # Receive message from SQS queue
+    response = sqs.receive_message(
+        QueueUrl=sqs_url,
+        AttributeNames=["SentTimestamp"],
+        MaxNumberOfMessages=10,
+        MessageAttributeNames=["All"],
+        VisibilityTimeout=30,
+        WaitTimeSeconds=1,
+    )
+
+    messages = response.get("Messages", [])
+    message_bodies = []
+
+    if messages:
 
-    copy_source = {"Bucket": bucket, "Key": key}
+        for message in messages:
+            message_bodies.append(message.get("Body"))
 
-    s3_resource.meta.client.copy(copy_source, bucket, f"archive/{key}")
-    logger.info(f"Archived file :: {key}")
+            receipt_handle = message["ReceiptHandle"]
 
-    # delete object from original bucket
-    s3_client.delete_object(Bucket=bucket, Key=key)
-    logger.info(f"Deleted file :: {key}")
+            # Delete received message from queue
+            sqs.delete_message(QueueUrl=sqs_url, ReceiptHandle=receipt_handle)
+            logger.info(f"Received and deleted message: {message}")
```

Comment:
Logging the entire message may become expensive and cause log parsing difficulties. Can we just show a single object instead (`key` or something)?

Reply:
Yeah, that makes sense, will change that too